### PR TITLE
Blog - Error Handling in Python

### PR DESCRIPTION
**Pull request by msali123**
Done with all checks.

### CODE REVIEW

**Comment by msali123** [comment]
[Preview Link](https://642ea8f20a8d0d21449ef959--ephemeral-narwhal-3ac3b2.netlify.app/blog/error-handling-in-python/)

@balapriyac

File: blog/_posts/2023-05-28-error-handling-in-python.md [hunk 1]
```diff
@@ -0,0 +1,389 @@
...
+      This is shown in the code below:
+
+      ~~~{.python caption="error-handling-in-python.ipynb"}
+      try:
+          with open('example.txt') as f:
+                  data = f.read()
+      except (FileNotFoundError, PermissionError) as e:
+          print(f"Error: {e}")
+      ~~~
+
```
Comment (balapriyac) [inline blog/_posts/2023-05-28-error-handling-in-python.md:111]:
```suggestion
      This is shown in the code below:

      ~~~{.python caption="error-handling-in-python.ipynb"}
      try:
          with open('example.txt') as f:
              data = f.read()
      except (FileNotFoundError, PermissionError) as e:
          print(f"Error: {e}")
      ~~~

```

File: blog/_posts/2023-05-28-error-handling-in-python.md [hunk 2]
```diff
@@ -0,0 +1,389 @@
...
+try:
+           with open("file.txt") as f:
+           content = f.read()
+       num = int("abc")
+except FileNotFoundError:
+    print("File not found!")
+except ValueError:
+    print("Invalid value!")
+~~~
+
```
Comment (balapriyac) [inline blog/_posts/2023-05-28-error-handling-in-python.md:145]:
```suggestion

~~~{.python caption="error-handling-in-python.ipynb"}
try:
    with open("file.txt") as f:
        content = f.read()
    num = int("abc")
except FileNotFoundError:
    print("File not found!")
except ValueError:
    print("Invalid value!")
~~~

```

File: blog/_posts/2023-05-28-error-handling-in-python.md [hunk 3]
```diff
@@ -0,0 +1,389 @@
...
+In the above example, the code in the try block attempts to read the contents of a file and convert a string to an integer. If the file is not found, the `FileNotFoundError` exception is raised and the corresponding except block is executed. Similarly, if the string `"abc"` cannot be converted to an integer, the `ValueError` exception is raised and the corresponding except block is executed.
+
+Here's another example that demonstrates how to handle multiple exceptions in a single try-except block:
+
+~~~{.python caption="error-handling-in-python.ipynb"}
+try:
+    with open("example.txt", "r") as file:
+            contents = file.read()
+    value = int(input("enter a number: "))
+    num = 10 / value
```
Comment (balapriyac) [inline blog/_posts/2023-05-28-error-handling-in-python.md:161]:
```suggestion
~~~{.python caption="error-handling-in-python.ipynb"}
try:
    with open("example.txt", "r") as file:
        contents = file.read()
    value = int(input("enter a number: "))
    num = 10 / value
```

**Review by balapriyac — CHANGES_REQUESTED** [review]

File: blog/_posts/2023-05-28-error-handling-in-python.md [hunk 4]
```diff
@@ -0,0 +1,389 @@
...
+In the above example, the `raise` statement is used to raise a `ValueError` exception if the value of `x` is greater than 5. When the `raise` statement is executed, a traceback is printed to the console along with the message associated with the exception.
+
+Python built-in exceptions like `ValueError`, `TypeError`, and `IndexError` can be explicitly raised using the `raise` statement. Here's an example to demonstrate this:
+
+~~~{.python caption="error-handling-in-python.ipynb"}
+def get_element(data, index):
+    if index >= len(data):
+           raise IndexError("Index out of range")
+    return data[index]
+~~~
```
Comment (balapriyac) [inline blog/_posts/2023-05-28-error-handling-in-python.md:231]:
```suggestion
~~~{.python caption="error-handling-in-python.ipynb"}
def get_element(data, index):
    if index >= len(data):
        raise IndexError("Index out of range")
    return data[index]
~~~
```

File: blog/_posts/2023-05-28-error-handling-in-python.md [hunk 5]
```diff
@@ -0,0 +1,389 @@
...
+file = None
+try:
+    file = open("file.txt")
+    # do some work with the file
+except:
+    print("An error occurred!")
+finally:
+    if file:
+           file.close()
+~~~
```
Comment (balapriyac) [inline blog/_posts/2023-05-28-error-handling-in-python.md:270]:
```suggestion
finally:
    if file:
         file.close()
~~~
```

File: blog/_posts/2023-05-28-error-handling-in-python.md [hunk 6]
```diff
@@ -0,0 +1,389 @@
...
+try:
+    file = open("file.txt")
+    # do some work with the file
+except:
+    print("An error occurred!")
+finally:
+    if file:
+          file.close()
+    if os.path.exists("file.txt"):
+           os.remove("file.txt")
```
Comment (balapriyac) [inline blog/_posts/2023-05-28-error-handling-in-python.md:288]:
```suggestion
    if file:
          file.close()
    if os.path.exists("file.txt"):
          os.remove("file.txt")
```

File: blog/_posts/2023-05-28-error-handling-in-python.md [hunk 7]
```diff
@@ -0,0 +1,389 @@
...
+By following these best practices, you can write more reliable, maintainable, and robust code that is easier to debug and understand.
+
+## Conclusion
+
+In conclusion, this article has provided an overview of error handling in Python, including the different types of errors that can occur, how to handle errors using the `try-except` block and the `finally` block, how to raise exceptions, and how to use assert statements for debugging. We also covered best practices for error handling, including writing clear and informative error messages, testing error handling code, using consistent error handling throughout your code, documenting error handling, and using logging to track errors.
+
+You should have a good understanding of how to handle errors in your Python projects and should feel confident in your ability to write more robust and reliable code.
+
+The code for this article can be found on [GitHub](https://github.com/Boadzie/error-handling-in-python)
```
Comment (balapriyac) [inline blog/_posts/2023-05-28-error-handling-in-python.md:387]:
```suggestion
The code for this article can be found on [GitHub](https://github.com/Boadzie/error-handling-in-python).
```

**Review by balapriyac — CHANGES_REQUESTED** [review]
Thanks, Sameer. Suggested a few changes.

**Comment by msali123** [comment]
@balapriyac 
[Changes Done](https://642ff5352eb8b01bbcdb303d--ephemeral-narwhal-3ac3b2.netlify.app/blog/error-handling-in-python/)

**Comment by balapriyac** [comment]
@jalletto this is ready to merge.

**Review by jalletto — APPROVED** [review]

**Pull request merged**